### PR TITLE
Backport RegexAction to DLP proto

### PR DESCRIPTION
**Pull request by elcasteel**
# Description

Backport change to add RegexAction to the API for DLP transformations.

# Context

Original PR: https://github.com/solo-io/gloo/pull/5283
API added to Envoy-gloo-ee: https://github.com/solo-io/envoy-gloo-ee/pull/185/files

### CODE REVIEW

**Review by sam-heilbron — DISMISSED** [review]

File: changelog/v1.9.0/regexaction-api.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+changelog:
```
Comment (sam-heilbron) [inline changelog/v1.9.0/regexaction-api.yaml:1]: this needs to go to the 1.8 directory

File: projects/gloo/api/v1/enterprise/options/dlp/dlp.proto [hunk 2]
```diff
@@ -186,12 +188,17 @@ message CustomAction {
     // This name is used for logging and debugging purposes.
     string name = 1;
     // The list of regex strings which will be applied in order.
-    repeated string regex = 2;
+    repeated string regex = 2 [deprecated = true];
```
Comment (sam-heilbron) [inline projects/gloo/api/v1/enterprise/options/dlp/dlp.proto:191]: I'm glad we're calling this out. Out of curiosity, does adding this flag have any tangible effect?

Comment (elcasteel) [reply projects/gloo/api/v1/enterprise/options/dlp/dlp.proto:191]: It doesn't affect the behavior.

**Pull request merged**